src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -128,6 +150,19 @@ public int stem(char s[], int len) {
     return rule23(s, len);
   }
 
+  /**
+   * Re-writes the char array based on an exception.
+   * @param s the terms char array
+   * @param exception the exceptions char array (must be smaller in length)
+   * @return 
+   */
+  private int handle_exception(char[] s, char[] exception) {
+    int len = min(s.length, exception.length);
```

Comment:
We added a test that makes sure that stems are always shorter/equal than the original words in the yaml exception list. Therefore we'll never get an ArrayOutOfBoundsException or similar as long as the tests are passing.